### PR TITLE
Fix bytecode hash checking for v3

### PR DESCRIPTION
**Pull request by nambrot**
### Description

PR adjusts the bytecode checking for V3 mailbox's

### Drive-by changes

Adds a violation when a contract is not proxied, but be expected to be

### CODE REVIEW

File: typescript/sdk/src/consts/bytecode.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 export enum BytecodeHash {
   MAILBOX_WITHOUT_LOCAL_DOMAIN_BYTE_CODE_HASH = '0x29b7294ab3ad2e8587e5cce0e2289ce65e12a2ea2f1e7ab34a05e7737616f457',
   MAILBOX_WITHOUT_LOCAL_DOMAIN_NONZERO_PAUSE_BYTE_CODE_HASH = '0x4e73e34c0982b93eebb4ac4889e9e4e1611f7c24feacf016c3a13e389f146d9c',
-  TRANSPARENT_PROXY_BYTECODE_HASH = '0x4dde3d0906b6492bf1d4947f667afe8d53c8899f1d8788cabafd082938dceb2d',
-  PROXY_ADMIN_BYTECODE_HASH = '0x7c378e9d49408861ca754fe684b9f7d1ea525bddf095ee0463902df701453ba0',
+  V3_MAILBOX_BYTECODE_HASH = '0x6e853444a6e38bb1d7ac7608b92a70cee83153c891c70ed882b2432134bb23a0',
+  TRANSPARENT_PROXY_BYTECODE_HASH = '0x320bda003dfa31828115be5c01b9f3e7eecaf2532afdb89d2b53559f2e7ab86d',
+  PROXY_ADMIN_BYTECODE_HASH = '0x13855ae57da3aadecb9259cecece16e1f434b8850fe95531f422e4e262f3f200',
```
Comment (yorhodes) [inline typescript/sdk/src/consts/bytecode.ts:6]: I guess this is without the compiler optimization enabled? we will need to change this

File: typescript/sdk/src/consts/bytecode.ts [hunk 2]
```diff
@@ -1,8 +1,9 @@
...
   INTERCHAIN_GAS_PAYMASTER_BYTECODE_HASH = '0xe995bcd732f4861606036357edb2a4d4c3e9b8d7e599fe548790ac1cf26888f8',
```
Comment (yorhodes) [inline typescript/sdk/src/consts/bytecode.ts:7]: I assume these are also stale

File: typescript/sdk/src/core/HyperlaneCoreChecker.ts [hunk 3]
```diff
@@ -103,27 +103,50 @@ export class HyperlaneCoreChecker extends HyperlaneAppChecker<
...
+      };
+      this.addViolation(violation);
+    } else {
+      await this.checkBytecode(
+        chain,
+        'Mailbox implementation',
+        implementation,
+        [
+          BytecodeHash.MAILBOX_WITHOUT_LOCAL_DOMAIN_BYTE_CODE_HASH,
+          BytecodeHash.MAILBOX_WITHOUT_LOCAL_DOMAIN_NONZERO_PAUSE_BYTE_CODE_HASH,
```
Comment (yorhodes) [inline typescript/sdk/src/core/HyperlaneCoreChecker.ts:123]: assume these should be removed?

File: typescript/sdk/src/core/HyperlaneCoreChecker.ts [hunk 4]
```diff
@@ -103,27 +103,50 @@ export class HyperlaneCoreChecker extends HyperlaneAppChecker<
...
+          BytecodeHash.V3_MAILBOX_BYTECODE_HASH,
+        ],
+        (bytecode) =>
+          // This is obviously super janky but basically we are searching
+          //  for the ocurrences of localDomain in the bytecode and remove
+          //  that to compare, but some coincidental ocurrences of
+          // localDomain in the bytecode should be not be removed which
+          // are just done via an offset guard
```
Comment (yorhodes) [inline typescript/sdk/src/core/HyperlaneCoreChecker.ts:131]: iiuc the immutables are always at the end of the bytecode (before the compiler metadata)
so maybe we should be a little more strict

**Review by yorhodes — APPROVED** [review]

File: typescript/sdk/src/consts/bytecode.ts (hunk 1 above)
Comment (nambrot) [reply typescript/sdk/src/consts/bytecode.ts:6]: Yes indeed

File: typescript/sdk/src/consts/bytecode.ts (hunk 2 above)
Comment (nambrot) [reply typescript/sdk/src/consts/bytecode.ts:7]: yup

**Pull request merged**